### PR TITLE
[ruff] enable PLW2901, redefined-loop-name

### PR DESCRIPTION
**Pull request by smackesey**
## Summary & Motivation

Turns on Ruff `PLW2901` (`redefined-loop-name`) and fixes ~30ish violations in our code. Typically this meant renaming the loop variable.

---

## How I Tested These Changes

BK

### CODE REVIEW

**Comment by smackesey** [comment]
Current dependencies on/for this PR:
* master
  * **PR #12889** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/12889" target="_blank">Graphite</a> 
    * **PR #12890** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/12890" target="_blank">Graphite</a>  👈
      * **PR #12891** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/12891" target="_blank">Graphite</a> 
        * **PR #12892** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/12892" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/12890?utm_source=stack-comment).

File: python_modules/dagster/dagster/_cli/asset.py [hunk 1]
```diff
@@ -157,8 +157,7 @@ def asset_wipe_command(key, **cli_args):
             confirmation = click.prompt(prompt)
 
         if confirmation == "DELETE":
-            with DagsterInstance.get() as instance:
-                instance.wipe_assets(asset_keys)
-                click.echo("Removed asset indexes from event logs")
+            instance.wipe_assets(asset_keys)
```
Comment (smackesey) [inline python_modules/dagster/dagster/_cli/asset.py:160]: Pretty sure this `DagsterInstance.get` is redundant, it is nested inside another one a few lines up.

File: python_modules/dagster/dagster/_core/definitions/asset_layer.py [hunk 2]
```diff
@@ -297,9 +297,6 @@ def asset_key_to_dep_node_handles(
             {}
         )  # memoized map of node output handles to all node output handle dependencies that are from ops
         for output_name, asset_key in assets_defs.keys_by_output_name.items():
-            output_def = assets_defs.node_def.output_def_named(output_name)
-            output_name = output_def.name
-
```
Comment (smackesey) [inline python_modules/dagster/dagster/_core/definitions/asset_layer.py:302]: Not sure exactly why this code is here as `output_name` appears redundant with `output_name` set in the loop. cc @clairelin135

File: python_modules/libraries/dagstermill/dagstermill_tests/test_ops.py [hunk 3]
```diff
@@ -342,7 +342,6 @@ def test_hello_world_reexecution():
             )
             reexecution_notebook_file.flush()
 
-            result = None
             reexecution_job = ReconstructablePipeline.for_file(
```
Comment (smackesey) [inline python_modules/libraries/dagstermill/dagstermill_tests/test_ops.py:345]: Doesn't seem to have a purpose and removing did not break test

File: python_modules/libraries/dagster-dbt/dagster_dbt_tests/cloud/utils.py [hunk 4]
```diff
@@ -91,7 +91,7 @@ def sample_runs_details(include_related=None, **kwargs):
                 "supports_docs": False,
                 "state": 10,
             }
-            run = deep_merge_dicts(run, kwargs)
```
Comment (smackesey) [inline python_modules/libraries/dagster-dbt/dagster_dbt_tests/cloud/utils.py:94]: The `deep_merge_dicts` here was not having its intended effect.

File: python_modules/dagster/dagster/_core/definitions/asset_layer.py (hunk 2 above)
Comment (clairelin135) [reply python_modules/dagster/dagster/_core/definitions/asset_layer.py:302]: yeah this makes sense to me! Agree that these lines are redundant, as long as bk tests still pass this should be fine 👍

Comment (smackesey) [reply python_modules/dagster/dagster/_core/definitions/asset_layer.py:302]: Cool-- actually mind reviewing this whole PR? It's the last one in the stack, alex did all the others

**Review by clairelin135 — APPROVED** [review]

**Comment by smackesey** [comment]
@smackesey queued this pull request to merge with [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/12890).

**Comment by smackesey** [comment]
@smackesey merged this pull request with [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/12890).

**Pull request merged**